addons/binding/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/SunSpecHandlerFactory.java

```diff
@@ -0,0 +1,114 @@
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SunSpecHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Nagy Attila Gabor - Initial contribution
+ */
+@Component(service = ThingHandlerFactory.class, immediate = true, configurationPid = "binding.sunspec", configurationPolicy = ConfigurationPolicy.OPTIONAL)
+public class SunSpecHandlerFactory extends BaseThingHandlerFactory {
+
+    /**
+     * Logger instance
+     */
+    @NonNull
+    private final Logger logger = LoggerFactory.getLogger(SunSpecHandlerFactory.class);
+
+    /**
+     * Reference to the modbus manager
+     */
+    private ModbusManager manager;
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = new HashSet<>();
+    static {
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_INVERTER_SINGLE_PHASE);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_INVERTER_SPLIT_PHASE);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_INVERTER_THREE_PHASE);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_METER_SINGLE_PHASE);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_METER_SPLIT_PHASE);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_METER_WYE_PHASE);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_METER_DELTA_PHASE);
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (manager == null) {
+            logger.debug("Modbus manager not found, can't continue");
+            return null;
+        }
+        if (thingTypeUID.equals(THING_TYPE_INVERTER_SINGLE_PHASE)
+                || thingTypeUID.equals(THING_TYPE_INVERTER_SPLIT_PHASE)
+                || thingTypeUID.equals(THING_TYPE_INVERTER_THREE_PHASE)) {
+            return new InverterHandler(thing, () -> manager);
+        } else if (thingTypeUID.equals(THING_TYPE_METER_SINGLE_PHASE)
+                || thingTypeUID.equals(THING_TYPE_METER_SPLIT_PHASE) || thingTypeUID.equals(THING_TYPE_METER_WYE_PHASE)
+                || thingTypeUID.equals(THING_TYPE_METER_DELTA_PHASE)) {
+            return new MeterHandler(thing, () -> manager);
+        }
+
+        return null;
+    }
+
+    /**
+     * Setter to accept the ModbusManager
+     *
+     * @param manager the modbus manager from org.openhab.io.transport.modbus package
+     */
+    @Reference(service = ModbusManager.class, cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.STATIC, unbind = "unsetManager")
```

Comment:
Please remove unbind (its implicit) and policy. The default is reasonable. MANDATORY is default.